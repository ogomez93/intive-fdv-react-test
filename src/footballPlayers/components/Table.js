import React from 'react';
import { connect } from 'react-redux';
import VisiblePlayersList from './VisiblePlayersList';
import './css/table.css';

const Table = () => (
  <table id="playersTable" className="custom-table w-full text-center text-xs border">
    <thead>
      <tr label="labels" className="text-white font-semibold">
        <td label="Player" className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark">Player</td>
        <td label="Position" className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark">Position</td>
        <td label="Team" className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark">Team</td>
        <td label="Age" className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark">Age</td>
      </tr>
    </thead>
    <VisiblePlayersList />
  </table>
);

export default connect()(Table);