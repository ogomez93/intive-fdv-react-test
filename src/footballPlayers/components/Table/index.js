import { branch, compose, renderComponent } from 'recompose';
import { connect } from 'react-redux';

import Table from './Table';
import Loader from '../Loader';

const mapStateToProps = state => ({
  fetching: state.footballPlayers.fetching
});

export default compose(
  connect(mapStateToProps),
  branch(
    ({ fetching }) => fetching,
    renderComponent(Loader)
  )
)(Table);
