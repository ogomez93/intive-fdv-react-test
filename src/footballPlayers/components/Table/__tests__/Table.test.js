import React from 'react';
import { shallow } from 'enzyme';

import Table from '../Table';

jest.mock('footballPlayers/components/VisiblePlayersList', () => 'VisiblePlayersList');

describe('Component: <Table />', () => {
  let wrapper;
  
  beforeAll(() => {
    wrapper = shallow(<Table />);
  });

  it('should render', () =>
    expect(wrapper.isEmptyRender()).toBe(false));

  it('should match the snapshot', () =>
    expect(wrapper).toMatchSnapshot());
  
  it('should render a table', () =>
    expect(wrapper.type()).toEqual('table'));

  describe('Labels', () => {
    describe('Player name', () => {
      let playerLabel;

      beforeAll(() => {
        playerLabel = wrapper.find('[label="Player"]');
      });

      it('should render the Player name label', () =>
        expect(playerLabel.text()).toEqual('Player'));
    });

    describe('Position', () => {
      let positionLabel;

      beforeAll(() => {
        positionLabel = wrapper.find('[label="Position"]');
      });

      it('should render the Position label', () =>
        expect(positionLabel.text()).toEqual('Position'));
    });

    describe('Team', () => {
      let teamLabel;

      beforeAll(() => {
        teamLabel = wrapper.find('[label="Team"]');
      });

      it('should render the Team label', () =>
        expect(teamLabel.text()).toEqual('Team'));
    });

    describe('Age', () => {
      let ageLabel;

      beforeAll(() => {
        ageLabel = wrapper.find('[label="Age"]');
      });

      it('should render the Age label', () =>
        expect(ageLabel.text()).toEqual('Age'));
    });
  });

  it('should render the VisiblePlayersList component', () =>
    expect(wrapper.find('VisiblePlayersList').isEmptyRender()).toEqual(false));
});
