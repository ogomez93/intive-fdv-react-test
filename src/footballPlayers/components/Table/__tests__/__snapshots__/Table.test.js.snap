// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Table /> should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Table />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <thead>
          <tr
                    className="text-white font-semibold"
                    label="labels"
          >
                    <td
                              className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                              label="Player"
                    >
                              Player
                    </td>
                    <td
                              className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                              label="Position"
                    >
                              Position
                    </td>
                    <td
                              className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                              label="Team"
                    >
                              Team
                    </td>
                    <td
                              className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                              label="Age"
                    >
                              Age
                    </td>
          </tr>
</thead>,
        <VisiblePlayersList />,
      ],
      "className": "custom-table w-full text-center text-xs border",
      "id": "playersTable",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <tr
            className="text-white font-semibold"
            label="labels"
>
            <td
                        className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                        label="Player"
            >
                        Player
            </td>
            <td
                        className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                        label="Position"
            >
                        Position
            </td>
            <td
                        className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                        label="Team"
            >
                        Team
            </td>
            <td
                        className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                        label="Age"
            >
                        Age
            </td>
</tr>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <td
                className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                label="Player"
>
                Player
</td>,
              <td
                className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                label="Position"
>
                Position
</td>,
              <td
                className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                label="Team"
>
                Team
</td>,
              <td
                className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                label="Age"
>
                Age
</td>,
            ],
            "className": "text-white font-semibold",
            "label": "labels",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Player",
                "className": "w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark",
                "label": "Player",
              },
              "ref": null,
              "rendered": "Player",
              "type": "td",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Position",
                "className": "w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark",
                "label": "Position",
              },
              "ref": null,
              "rendered": "Position",
              "type": "td",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Team",
                "className": "w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark",
                "label": "Team",
              },
              "ref": null,
              "rendered": "Team",
              "type": "td",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Age",
                "className": "w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark",
                "label": "Age",
              },
              "ref": null,
              "rendered": "Age",
              "type": "td",
            },
          ],
          "type": "tr",
        },
        "type": "thead",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "VisiblePlayersList",
      },
    ],
    "type": "table",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <thead>
            <tr
                        className="text-white font-semibold"
                        label="labels"
            >
                        <td
                                    className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                                    label="Player"
                        >
                                    Player
                        </td>
                        <td
                                    className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                                    label="Position"
                        >
                                    Position
                        </td>
                        <td
                                    className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                                    label="Team"
                        >
                                    Team
                        </td>
                        <td
                                    className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                                    label="Age"
                        >
                                    Age
                        </td>
            </tr>
</thead>,
          <VisiblePlayersList />,
        ],
        "className": "custom-table w-full text-center text-xs border",
        "id": "playersTable",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <tr
              className="text-white font-semibold"
              label="labels"
>
              <td
                            className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                            label="Player"
              >
                            Player
              </td>
              <td
                            className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                            label="Position"
              >
                            Position
              </td>
              <td
                            className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                            label="Team"
              >
                            Team
              </td>
              <td
                            className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                            label="Age"
              >
                            Age
              </td>
</tr>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <td
                  className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                  label="Player"
>
                  Player
</td>,
                <td
                  className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                  label="Position"
>
                  Position
</td>,
                <td
                  className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                  label="Team"
>
                  Team
</td>,
                <td
                  className="w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark"
                  label="Age"
>
                  Age
</td>,
              ],
              "className": "text-white font-semibold",
              "label": "labels",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Player",
                  "className": "w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark",
                  "label": "Player",
                },
                "ref": null,
                "rendered": "Player",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Position",
                  "className": "w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark",
                  "label": "Position",
                },
                "ref": null,
                "rendered": "Position",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Team",
                  "className": "w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark",
                  "label": "Team",
                },
                "ref": null,
                "rendered": "Team",
                "type": "td",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Age",
                  "className": "w-1/4 border-2 border-black px-1 py-3 sm:py-6 bg-grey-dark",
                  "label": "Age",
                },
                "ref": null,
                "rendered": "Age",
                "type": "td",
              },
            ],
            "type": "tr",
          },
          "type": "thead",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "VisiblePlayersList",
        },
      ],
      "type": "table",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
